core/Pos/Util/Util.hs

```diff
@@ -577,6 +579,12 @@ withTempFile tmpDir template action =
      ignoringIOErrors :: MC.MonadCatch m => m () -> m ()
      ignoringIOErrors ioe = ioe `MC.catch` (\e -> const (return ()) (e :: Prelude.IOError))
 
+applyPattern :: Show a => FilePath -> a -> FilePath
```

Comment:
Why is it here and not documented? This function seems to do something weird which is used only by keygen, so it should probably be somewhere near keygen.
If you really think it's general enough, it should be at least documented.